tools/alara_output_processing.py

```diff
@@ -0,0 +1,399 @@
+            DataFrames with associated metadata containing ALARA output data
+            for different variables. Allows for processing of either existing
+            DataFrames, with the requirement that the user provide the
+            relevant data source, evaulated variable, and its relevant units
+            to be packaged into a data dictionary. Alternatively, this
+            function can directly read in an ALARA output file and parse all
+            tables and their metadata internally.
+
+        Arguments:
+            data_source (dict or pandas.core.frame.DataFrame): ALARA output
+                data. If parsing directly from ALARA output files, data_source
+                must be formatted as a dictionary of the form:
+                data_source = {
+                    Data Library 1 : path/to/output/file/for/data/library1,
+                    Data Library 2 : path/to/output/file/for/data/library2
+                }
+                If processing a preexisting DataFrame, then data_source is
+                just the DataFrame itself.
+            inp_datalib (str or None, optional): Data source of the selected
+                DataFrame. Required if processing a preexisting DataFrame;
+                irrelevant if parsing directly from ALARA output files.
+                (Defaults to None)
+            inp_variable (str or None, optional): Evaluated variable for the
+                selected DataFrame. Required if processing a preexisting
+                DataFrame; irrelevant if parsing directly from ALARA output
+                files.
+                (Defaults to None)
+            inp_unit (str or None, optional): Appropriate unit for the
+                evaluated variable for the selected DataFrame. Required if
+                processing a preexisting DataFrame; irrelevant if parsing
+                directly from ALARA output files.
+                (Defaults to None)
+                
+        Returns:
+            dfs (list of dicts): List of dictionaries containing ALARA output
+                DataFrames and their metadata, of the form:
+                df_dict = {
+                    'Data Source' : (Either 'fendl2' or 'fendl3'),
+                    'Variable'    : (Any ALARA output variable, dependent on
+                                    ALARA run parameters),
+                    'Unit'        : (Respective unit matching the above
+                                    variable),
+                    'Data'        : (DataFrame containing ALARA output data
+                                    for the given data source, variable, and
+                                    unit)
+                }
+        '''
+        
+        dfs = {}
+
+        if isinstance(data_source, pd.DataFrame):
+            dfs.update(
+                cls.make_entry(
+                    inp_datalib, inp_variable, inp_unit, data_source
+                )
+            )
+            return dfs
```

Comment:
Maybe we just have users call `make_entry` directly in these cases, and only use this method for processing multiple output files?  Then we could have one method called `make_entry` and one called `make_entries`? Or something like that...